### DISCUSSION TITLE
How to highlight a specified NVDA object by code?

### DISCUSSION THREAD

**Discussion by hwf1324** [nvaccess/nvda#17054] [Q&A]
I have an add-on ([Object Viewer](https://github.com/hwf1324/objectViewer)) that uses a GUI to display the structure and information of NVDAObjects. Like Microsoft's Accessibility Insights For Windows, I want the selected NVDAObject to be highlighted on the screen.

Can I do this with the existing `NVDAHighlighter` related code? It seems that it can only be called through a limited number of `Context`s?

**Comment by hwf1324** [comment]
CC: @LeonarddeR

**Comment by CyrilleB79** [comment]
I have not tested anything myself.

Have you tried to create a custom class inheriting from `HighlightWindow` (in `NVDAHighlighter.py`) and to redefine the `_paint` method?
